### PR TITLE
taras_krasitskyi - class Product

### PR DESCRIPTION
**Pull request by Taras-Kr**

### CODE REVIEW

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 1]
```diff
@@ -0,0 +1,116 @@
...
+import java.util.List;
+import java.util.stream.Collectors;
+
+import static com.codeborne.selenide.Selenide.*;
+
+@RequiredArgsConstructor
+public class Product {
+
+    @NonNull
+    private final String xPath;
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:19]: why is there a need to pass xpath in constructor?

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 2]
```diff
@@ -0,0 +1,116 @@
...
+        return productTitle;
+    }
+
+    @Step("Product: get promo label title")
+    public String getPromoLabelTitle() {
+        try {
+            promoLabelTitle = $x(String.format("%s%s", xPath, "//span[contains(@class,'goods-tile__label')]")).text();
+        } catch (AssertionError exception) {
+            promoLabelTitle = StringUtils.EMPTY;
+        }
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:41]: do not use try..catch to check simple conditions
use if el.exist() then return elemen title else return empty

Comment (toleksyn) [reply src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:41]: same for all

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 3]
```diff
@@ -0,0 +1,116 @@
...
+
+    @Step("Product: get available colors")
+    public List<String> getAvailableColors() {
+        try {
+            availableColors = $$x(String.format("%s%s", xPath, "//span[@class='goods-tile__colors-content']"))
+                    .stream()
+                    .map(element -> Color.fromString(element.getCssValue("background-color")).asHex())
+                    .collect(Collectors.toList());
+        } catch (AssertionError exception) {
+            availableColors = null;
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:53]: returning null is a code smell, return empty list instead

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 4]
```diff
@@ -0,0 +1,116 @@
...
+        }
+        return availableColors;
+    }
+
+    @Step("Product: get product old price")
+    public BigDecimal getProductOldPrice() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:59]: drop "product" from method names, it'll be obvious when use: product.getOldPrice()
same for all

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 5]
```diff
@@ -0,0 +1,116 @@
...
+        try {
+            availability = $x(String.format("%s%s", xPath, "//div[contains(@class,'goods-tile__availability')]")).text();
+        } catch (AssertionError exception) {
+            availability = StringUtils.EMPTY;
+        }
+        return availability;
+    }
+
+    @Step("Product: get amount reviews")
+    public int getAmountReviews() {
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:95]: getReviewAmount()

**Review by toleksyn — CHANGES_REQUESTED** [review]

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java (hunk 1 above)
Comment (Taras-Kr) [reply src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:19]: each Product on the ProductTypePage has its xpath (difference by index in the elements collection $$x(//div[@class='goods-tile__inner').
when I want to get Product by index from the collection, I use
public Product getProduct(int numberProduct) {
    return new Product(String.format(("(//div[@class='goods-tile__inner'])[%s]"), numberProduct))
} from ProductTypePage
xpath from Product uses in the get* methods for locate product elements(such as title, price....)

File: src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+        productTitle = $x(String.format("%s%s", xPath, "//span[@class='goods-tile__title']")).text();
+        return productTitle;
+    }
+
+    @Step("Product: get promo label title")
+    public String getPromoLabelTitle() {
+        if($x(String.format("%s%s", xPath, "//span[contains(@class,'goods-tile__label')]")).exists()){
+            promoLabelTitle = $x(String.format("%s%s", xPath, "//span[contains(@class,'goods-tile__label')]")).text();
+        }else{
+            promoLabelTitle = StringUtils.EMPTY;
```
Comment (toleksyn) [inline src/main/java/com/ss/ita/rozetka/pageobject/elements/Product.java:41]: code format
don't duplicate
same for all

**Review by toleksyn — CHANGES_REQUESTED** [review]

**Pull request merged**